kernel/src/process_checker/signature.rs

```diff
@@ -0,0 +1,268 @@
+    ) -> Result<(), (ErrorCode, TbfFooterV2Credentials, &'static [u8])> {
+        self.credentials.set(credentials);
+
+        if credentials.format() == self.credential_type {
+            // Save the signature we are trying to compare with.
+            self.signature.map(|b| {
+                b.as_mut_slice()[..SL].copy_from_slice(&credentials.data()[..SL]);
+            });
+
+            // Add the process binary to compute the hash.
+            self.hasher.clear_data();
+            match self.hasher.add_data(SubSlice::new(binary)) {
+                Ok(()) => Ok(()),
+                Err((e, b)) => Err((e, credentials, b.take())),
+            }
+        } else {
+            Err((ErrorCode::NOSUPPORT, credentials, binary))
+        }
+    }
+
+    fn set_client(&self, client: &'static dyn Client<'static>) {
+        self.client.replace(client);
+    }
+}
+
+impl<
+        'a,
+        S: hil::public_key_crypto::signature::SignatureVerify<'static, HL, SL>,
+        H: hil::digest::DigestDataHash<'a, HL>,
+        const HL: usize,
+        const SL: usize,
+    > AppUniqueness for AppCheckerSignature<'a, S, H, HL, SL>
+{
+    fn different_identifier(&self, process_a: &dyn Process, process_b: &dyn Process) -> bool {
+        let cred_a = process_a.get_credentials();
+        let cred_b = process_b.get_credentials();
+
+        // If it doesn't have credentials, it is by definition
+        // different. It should not be runnable (this checker requires
+        // credentials), but if this returned false it could block
+        // runnable processes from running.
+        cred_a.map_or(true, |a| {
+            cred_b.map_or(true, |b| {
+                // Two IDs are different if they have a different format,
+                // different length (should not happen, but worth checking for
+                // the next test), or any byte of them differs.
+                if a.format() != b.format() {
+                    true
+                } else if a.data().len() != b.data().len() {
+                    true
+                } else {
+                    for (aval, bval) in a.data().iter().zip(b.data().iter()) {
+                        if aval != bval {
+                            return true;
+                        }
+                    }
+                    false
+                }
+            })
+        })
+    }
+}
+
+impl<
+        'a,
+        S: hil::public_key_crypto::signature::SignatureVerify<'static, HL, SL>,
+        H: hil::digest::DigestDataHash<'a, HL>,
+        const HL: usize,
+        const SL: usize,
+    > Compress for AppCheckerSignature<'a, S, H, HL, SL>
+{
+    fn to_short_id(&self, _process: &dyn Process, credentials: &TbfFooterV2Credentials) -> ShortID {
+        let data = credentials.data();
+        if data.len() < 4 {
+            // Should never trigger, as we only approve signature credentials.
+            return ShortID::LocallyUnique;
+        }
+        let id: u32 = 0x8000000_u32
```

Comment:
Since this is exactly duplicated from `basic.rs` and is a general transformation of a global to short id, this probably ought to live in a utility function in the parent mod or something, and this whole function would be, e.g.:

```rust
return super::to_short_id(credentials.data());
```

Reply:
Yes, and it will, I just can't keep all the PRs totally rebased on top of each other. This is the cost of the pace.

https://github.com/tock/tock/pull/3849/files#diff-1c6e456a9fcd5d680418f9784fe564e00921c5d2e6c76d49a8784b20ac1df936R168 will split policy from appid assignment, and we won't have to duplicate compress implementations.

Reply:
This is also discussed here: https://github.com/tock/tock/issues/3813 and planned here: https://github.com/tock/tock/issues/3876